src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -14,18 +14,22 @@ object Interpret {
 
   def apply[T](ir: IR): T = apply(ir, Env.empty[(Any, Type)], IndexedSeq(), None).asInstanceOf[T]
 
-  def apply[T](ir: IR,
+  def apply[T](ir0: IR,
     env: Env[(Any, Type)],
     args: IndexedSeq[(Any, Type)],
     agg: Option[Agg]): T = {
 
-    log.info("interpret:\n" + Pretty(ir))
 
     val (typeEnv, valueEnv) = env.m.foldLeft((Env.empty[Type], Env.empty[Any])) {
       case ((e1, e2), (k, (value, t))) => (e1.bind(k, t), e2.bind(k, value))
     }
 
+    var ir = ir0
+    ir = Optimize(ir)
```

Comment:
should this be parameterized? If we want to use the interpreter to test our compiler and optimizer, then we need a way to take the stupid path

Reply:
Yeah, agreed, will change.

Reply:
🤔 🕐

Reply:
Don't ask :-/